File: algorithms/121.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock


class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        min_price = sys.maxsize
        max_profit = 0
        for p in prices:
            min_price = min(p, min_price)
            max_profit = max(p - min_price, max_profit)
        return max_profit

File: algorithms/121.swift
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock

class Solution {
    func maxProfit(prices: [Int]) -> Int {
        var minPrice = Int.max
        var maxProfit = 0
        for p in prices {
            minPrice = min(p, minPrice)
            maxProfit = max(p-minPrice, maxProfit)
        }
        return maxProfit
    }
}
